types/openrct2/index.d.ts

```diff
@@ -0,0 +1,2456 @@
+         * @param action The name of the action.
+         * @param args The action parameters.
+         * @param callback The function to be called with the result of the action.
+         */
+        executeAction(action: ActionType, args: object, callback: (result: GameActionResult) => void): void;
+        executeAction(action: string, args: object, callback: (result: GameActionResult) => void): void;
+
+        /**
+         * Subscribes to the given hook.
+         */
+        subscribe(hook: HookType, callback: Function): IDisposable;
+
+        subscribe(hook: "action.query", callback: (e: GameActionEventArgs) => void): IDisposable;
+        subscribe(hook: "action.execute", callback: (e: GameActionEventArgs) => void): IDisposable;
+        subscribe(hook: "interval.tick", callback: () => void): IDisposable;
+        subscribe(hook: "interval.day", callback: () => void): IDisposable;
+        subscribe(hook: "network.chat", callback: (e: NetworkChatEventArgs) => void): IDisposable;
+        subscribe(hook: "network.authenticate", callback: (e: NetworkAuthenticateEventArgs) => void): IDisposable;
+        subscribe(hook: "network.join", callback: (e: NetworkEventArgs) => void): IDisposable;
+        subscribe(hook: "network.leave", callback: (e: NetworkEventArgs) => void): IDisposable;
+        subscribe(hook: "ride.ratings.calculate", callback: (e: RideRatingsCalculateArgs) => void): IDisposable;
+        subscribe(hook: "action.location", callback: (e: ActionLocationArgs) => void): IDisposable;
+
+        /**
+         * Registers a function to be called every so often in realtime, specified by the given delay.
+         * @param callback The function to call every time the delay has elapsed.
+         * @param delay The number of milliseconds to wait between each call to the given function.
+         */
+        setInterval(callback: Function, delay: number): number;
+
+        /**
+         * Like `setInterval`, except the callback will only execute once after the given delay.
+         * @param callback The function to call after the given delay has elapsed.
+         * @param delay The number of milliseconds to wait for before calling the given function.
+         */
+        setTimeout(callback: Function, delay: number): number;
+
+        /**
+         * Removes the registered interval specified by the numeric handle. The handles
+         * are shared with `setTimeout`.
+         * @param handle
+         */
+        clearInterval(handle: number): void;
+
+        /**
+         * Removes the registered timeout specified by the numeric handle. The handles
+         * are shared with `setInterval`.
+         * @param handle The numerical handle of the registered timeout to remove.
+         */
+        clearTimeout(handle: number): void;
+    }
+
+    interface Configuration {
+        getAll(namespace: string): { [name: string]: any };
+        get<T>(key: string): T | undefined;
+        get<T>(key: string, defaultValue: T): T;
```

Comment:
I do think these might be better off as `unknown`; would that be acceptable for most users?

Reply:
If plugins are already using this, e.g. `let x = get<number>("language", 0);`, would that plugin then fail to build if the definition was changed upstream?